rcnn/rcnn.cpp

```diff
@@ -63,40 +69,39 @@ const std::vector<float>& aspect_ratios) {
     return res;
 }
 
-// transpose && resize && normalization && padding
-ITensor* DataPreprocess(INetworkDefinition *network, ITensor& input) {
-    // get h and w
-    auto input_hw = input.getDimensions();
-    int c = input_hw.d[2];
-    int height = input_hw.d[0];
-    int width = input_hw.d[1];
-
-    // resize
-    float ratio = MIN_SIZE / static_cast<float>(std::min(height, width));
-    float newh = 0, neww = 0;
-    if (height < width) {
-        newh = MIN_SIZE;
-        neww = ratio * width;
+cv::Mat preprocessImg_(cv::Mat& img, int input_w, int input_h) {
+    int w, h, x, y;
+    float r_w = input_w / (img.cols*1.0);
+    float r_h = input_h / (img.rows*1.0);
+    if (r_h > r_w) {
+        w = input_w;
+        h = r_w * img.rows;
+        x = 0;
+        y = (input_h - h) / 2;
     } else {
-        newh = ratio * height;
-        neww = MIN_SIZE;
-    }
-    if (std::max(newh, neww) > MAX_SIZE) {
-        ratio = MAX_SIZE / static_cast<float>(std::max(newh, neww));
-        newh = newh * ratio;
-        neww = neww * ratio;
+        w = r_h * img.cols;
+        h = input_h;
+        x = (input_w - w) / 2;
+        y = 0;
     }
-    height = static_cast<int>(newh + 0.5);
-    width = static_cast<int>(neww + 0.5);
-    auto resize_layer = network->addResize(input);
-    assert(resize_layer);
-    resize_layer->setResizeMode(ResizeMode::kLINEAR);
-    resize_layer->setOutputDimensions(Dims3{ height, width, c });
-    IMAGE_HEIGHT = height;
-    IMAGE_WIDTH = width;
+    cv::Mat re(h, w, CV_8UC3);
+    cv::resize(img, re, re.size(), 0, 0, cv::INTER_LINEAR);
+    cv::Mat out(input_h, input_w, CV_8UC3, cv::Scalar(128, 128, 128));
```

Comment:
looks like you do the center-padding with 128, but rcnn-c4 dont need padding and rcnn-fpn use lower-right padding with 0 (i remember)

Reply:
The padding operation is imported from the function preprocessImg() in your previous common.hpp file, so you actually have used padding in rcnn-c4.

Reply:
in fact it's not really preprocessing, its purpose is just make sure the input be same size, so that no matter what is the size of original image, they will be same when they are feed to the model. The really preprocess is ITensor* DataPreprocess(INetworkDefinition *network, ITensor& input) which is inserted into trt engine.
so, the ratio of the postprocess will be equal

Reply:
hah, the main purpose is simple (outputting detections and segmentations on the original images, not the padded images).